### DISCUSSION TITLE
Converting dataset with multiple CRS to COG

### DISCUSSION THREAD

**Discussion by delbronski** [developmentseed/titiler#997] [Q&A]
I have a dataset of 700+ files. 

Here is some data about a single file:

COG: True
Compression: LZW
ColorSpace: None

```
Profile
    Width:            20686
    Height:           89414
    Bands:            1
    Tiled:            True
    Dtype:            uint8
    NoData:           0.0
    Alpha Band:       False
    Internal Mask:    False
    Interleave:       BAND
    ColorMap:         False
    ColorInterp:      ('gray',)
    Scales:           (1.0,)
    Offsets:          (0.0,)

Geo
    Crs:              EPSG:32702
    Origin:           (396570.0, 2011064.3450437225)
    Resolution:       (10.0, -10.0)
    BoundingBox:      (396570.0, 1116924.3450437225, 603430.0, 2011064.3450437225)
    MinZoom:          5
    MaxZoom:          12

Image Metadata
    AREA_OR_POINT: Area

Image Structure
    COMPRESSION: LZW
    INTERLEAVE: BAND

Band 1
    ColorInterp: gray

IFD
    Id      Size           BlockSize     Decimation           
    0       20686x89414    512x512       0
```

I used gdalbuildvrt followed by gdal_translate to create a single COG. The process took around 6 hours. However, the final output looked nothing like the original dataset. This is a map of the whole world, but only a small portion was shown. 

One thing I noticed about the dataset is that there are many different CRSs within the dataset. Every 10-15 files the CRS changes. I suspect this has something to do with it. 

Any suggestions for how to take 700 files (32gb total size) belonging to the same dataset with different CRS and make one big cog out of it?

**Comment by vincentsarago** [comment]
did you used `-allow_projection_difference` option when building the VRT? 
did you assigned an output CRS to the VRT?

**Reply by delbronski** [reply]
Hey Vincent, 

Thanks for the tip. I just read about **-allow_projection_difference**.  I did not see that before. 

So I see that when specifying both of these options no actual reprojection is done. 

<img width="1002" alt="Screenshot 2024-10-16 at 15 18 29" src="https://github.com/user-attachments/assets/681063a2-eb5f-476d-83ea-b06fe4410cb7">

I'm going to try this and then run the gdal_translate command on the VRT and see what I get.
